integration/spark/src/main/common/java/io/openlineage/spark/agent/facets/builder/EnvironmentFacetBuilder.java

```diff
@@ -0,0 +1,119 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.spark.scheduler.SparkListenerEvent;
+
+public class EnvironmentFacetBuilder
+    extends CustomFacetBuilder<SparkListenerEvent, EnvironmentFacet> {
+  private static HashMap<String, Object> dbProperties;
+  private static final org.slf4j.Logger log =
+      org.slf4j.LoggerFactory.getLogger(EnvironmentFacetBuilder.class);
+  private final OpenLineageContext openLineageContext;
+
+  public EnvironmentFacetBuilder(OpenLineageContext openLineageContext) {
+    this.openLineageContext = openLineageContext;
+  }
+
+  @Override
+  protected void build(
+      SparkListenerEvent event, BiConsumer<String, ? super EnvironmentFacet> consumer) {
+    if (System.getenv().containsKey("DATABRICKS_RUNTIME_VERSION")) {
+      consumer.accept(
+          "environment-properties",
+          new EnvironmentFacet(getDatabricksEnvironmentalAttributes(event)));
+    }
+  }
+
+  private HashMap<String, Object> getDatabricksEnvironmentalAttributes(
+      SparkListenerEvent jobStart) {
+    dbProperties = new HashMap<>();
+    String urlString =
+        "http://"
+            + openLineageContext
+                .getProperties()
+                .getProperty("spark.databricks.clusterUsageTags.driverInstancePrivateIp")
+            + ":7070/?type=%22com.databricks.backend.daemon.data.common.DataMessages$GetMountsV2%22";
+
+    List<String> dbPropertiesKeys =
+        Arrays.asList(
+            "orgId",
+            "spark.databricks.clusterUsageTags.clusterOwnerOrgId",
+            "spark.databricks.notebook.path",
+            "spark.databricks.job.type",
+            "spark.databricks.job.id",
+            "spark.databricks.job.runId",
+            "user",
+            "userId",
+            "spark.databricks.clusterUsageTags.clusterName",
+            "spark.databricks.clusterUsageTags.azureSubscriptionId");
+
+    dbPropertiesKeys.stream()
+        .forEach(
+            (p) -> {
+              dbProperties.put(p, openLineageContext.getProperties().getProperty(p));
+            });
+
+    dbProperties.put("mountPoints", getDatabricksMountpoints(urlString));
+
+    return dbProperties;
+  }
+
+  private static List<Map<String, String>> getDatabricksMountpoints(String urlString) {
+    List<Map<String, String>> databricksMountpoints = new ArrayList<>();
+
+    try {
+      String result = "";
+      HttpPost post = new HttpPost(urlString);
+      post.addHeader("Sessionid", "1234");
```

Comment:
Do we really want to put `1234` here?

Reply:
This could be any random number, so simply I added 1234